From 25e8a6affad663e18558c3267b67b217c5a032c3 Mon Sep 17 00:00:00 2001
From: Ding Yunliang <yunliang.ding@intel.com>
Date: Fri, 15 Jun 2018 14:57:27 +0800
Subject: [PATCH 102/129] media: intel-ipu4: depracated abi sanity check

Message for Open Source:
The abi check is now having heavy maintain burden, and issue was not
easy to find, also makes code hard to read. So move this check
outside of kernel space, as this is an internal match check,
should not bother kernel users.

Message for Internal:
[Issue/Feature]
[Root Cause/Changes]
Move to icg-sw-devtools.

Change-Id: I471304d10a6dd92636448ea8f82f553044d27894
Tracked-On: #JICCCN-228
Signed-off-by: Ding Yunliang <yunliang.ding@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
---
 drivers/media/pci/intel/ici/libintel-ipu4_ici.c    |  14 --
 .../media/pci/intel/ipu4/ipu4-css/Makefile.isyslib |   1 -
 .../ipu4/ipu4-css/lib2600psys/libcsspsys2600.c     | 150 ++-----------------
 .../media/pci/intel/ipu4/ipu4-css/libintel-ipu4.c  |  14 --
 .../pci/intel/ipu4/ipu4p-css/Makefile.isyslib      |   1 -
 .../ipu4/ipu4p-css/lib2600psys/libcsspsys2600.c    | 150 ++-----------------
 .../pci/intel/ipu4/ipu4p-css/libintel-ipu4p.c      |  14 --
 drivers/media/pci/intel/libintel-checker.c         | 165 ---------------------
 drivers/media/pci/intel/libintel-checker.h         |   9 --
 9 files changed, 32 insertions(+), 486 deletions(-)
 delete mode 100644 drivers/media/pci/intel/libintel-checker.c
 delete mode 100644 drivers/media/pci/intel/libintel-checker.h

diff --git a/drivers/media/pci/intel/ici/libintel-ipu4_ici.c b/drivers/media/pci/intel/ici/libintel-ipu4_ici.c
index cc31663..087e657 100644
--- a/drivers/media/pci/intel/ici/libintel-ipu4_ici.c
+++ b/drivers/media/pci/intel/ici/libintel-ipu4_ici.c
@@ -7,7 +7,6 @@
 #include "ici/ici-isys.h"
 #include "ipu-wrapper.h"
 #include <ia_css_isysapi.h>
-#include "libintel-checker.h"
 
 #define ipu_lib_call_notrace_unlocked(func, isys, ...)		\
 	({								\
@@ -387,19 +386,6 @@ int ipu_fw_isys_complex_cmd(struct ici_isys *isys,
 }
 EXPORT_SYMBOL_GPL(ipu_fw_isys_complex_cmd);
 
-static int __init library_init(void)
-{
-	ipu_isys_abi_checker();
-	return 0;
-}
-
-static void __exit library_exit(void)
-{
-}
-
-module_init(library_init);
-module_exit(library_exit);
-
 MODULE_LICENSE("GPL");
 MODULE_DESCRIPTION("Intel ipu library");
 EXPORT_SYMBOL_GPL(ia_css_isys_device_open);
diff --git a/drivers/media/pci/intel/ipu4/ipu4-css/Makefile.isyslib b/drivers/media/pci/intel/ipu4/ipu4-css/Makefile.isyslib
index 2659e9b..c99c9a8 100644
--- a/drivers/media/pci/intel/ipu4/ipu4-css/Makefile.isyslib
+++ b/drivers/media/pci/intel/ipu4/ipu4-css/Makefile.isyslib
@@ -17,7 +17,6 @@ $(shell cp -f $(srcpath)/$(src)/../ipu-wrapper.c $(srcpath)/$(src)/ipu4-css/ipu-
 
 intel-ipu4-isys-csslib-objs := \
 			ipu4-css/libintel-ipu4.o \
-			../libintel-checker.o \
 			$(IPU_ISYSLIB_SRC)
 
 ifeq ($(CONFIG_VIDEO_INTEL_IPU), m)
diff --git a/drivers/media/pci/intel/ipu4/ipu4-css/lib2600psys/libcsspsys2600.c b/drivers/media/pci/intel/ipu4/ipu4-css/lib2600psys/libcsspsys2600.c
index 9b66bc8..ee13b8e 100644
--- a/drivers/media/pci/intel/ipu4/ipu4-css/lib2600psys/libcsspsys2600.c
+++ b/drivers/media/pci/intel/ipu4/ipu4-css/lib2600psys/libcsspsys2600.c
@@ -13,135 +13,21 @@
 
 #include <linux/delay.h>
 #include <linux/module.h>
-#include "libcsspsys2600.h"
-#include <vied_nci_psys_resource_model.h>
-#include <ia_css_psys_device.h>
-#include <ipu_device_cell_properties_func.h>
 
 #include <uapi/linux/ipu-psys.h>
 
 #include "ipu.h"
-#include "ipu-fw-psys.h"
+#include "ipu-mmu.h"
 #include "ipu-psys.h"
+#include "ipu-fw-psys.h"
 #include "ipu-wrapper.h"
-#include "ipu-mmu.h"
+#include "libcsspsys2600.h"
 
-#include <ia_css_psys_process_group_cmd_impl.h>
-#include <ia_css_psys_process_private_types.h>
-#include <ia_css_psys_init.h>
-#include <ia_css_psys_transport.h>
-#include <ia_css_terminal_base_types.h>
-#include <ia_css_terminal_types.h>
-#include <ia_css_program_group_param_types.h>
-#include <ia_css_psys_terminal_private_types.h>
-#include <ia_css_program_group_data.h>
+#include <vied_nci_psys_resource_model.h>
+#include <ia_css_psys_device.h>
+#include <ipu_device_cell_properties_func.h>
 #include <ia_css_psys_program_group_private.h>
-
-#define BASIC_ABI_CHECK
-
-#ifndef BASIC_ABI_CHECK
-#define ABI_CHECK(a, b, field)					\
-	{								\
-	if (offsetof(typeof(*a), field) != offsetof(typeof(*b), field))	\
-		pr_err("intel_ipu4 psys ABI mismatch %s\n",		\
-		       __stringify(field));		    \
-	}
-#else
-#define ABI_CHECK(a, b, field) { }
-#endif
-
-#define SIZE_OF_CHECK(a, b) \
-	{		    \
-	if (sizeof(*a) != sizeof(*b))\
-		pr_err("intel_ipu4 psys ABI size of mismatch %s\n",	\
-		       __stringify(a));					\
-	}								\
-
-static void abi_sanity_checker(void)
-{
-	struct ipu_fw_psys_process_group *ipu_fw_psys_pg;
-	struct ia_css_process_group_s *ia_css_pg;
-
-	struct ipu_fw_psys_process *ipu_fw_psys_ps;
-	struct ia_css_process_s *ia_css_ps;
-
-	struct ipu_fw_psys_srv_init *ipu_fw_psys_init;
-	struct ia_css_psys_server_init *ia_css_psys_init;
-
-	struct ipu_fw_psys_cmd *ipu_fw_psys_cmd;
-	struct ia_css_psys_cmd_s *ia_css_psys_cmd;
-
-	struct ipu_fw_psys_event *ipu_fw_psys_event;
-	struct ia_css_psys_event_s *ia_css_psys_event;
-
-	struct ipu_fw_psys_terminal *ipu_fw_psys_terminal;
-	struct ia_css_terminal_s *ia_css_terminal;
-
-	struct ipu_fw_psys_param_terminal *ipu_fw_psys_param_terminal;
-	struct ia_css_param_terminal_s *ia_css_param_terminal;
-
-	struct ipu_fw_psys_param_payload *ipu_fw_psys_param_payload;
-	struct ia_css_param_payload_s *ia_css_param_payload;
-
-	struct ipu_fw_psys_data_terminal *ipu_fw_psys_data_terminal;
-	struct ia_css_data_terminal_s *ia_css_data_terminal;
-
-	struct ipu_fw_psys_frame *ipu_fw_psys_frame;
-	struct ia_css_frame_s *ia_css_frame;
-
-	struct ipu_fw_psys_frame_descriptor *ipu_fw_psys_frame_descriptor;
-	struct ia_css_frame_descriptor_s *ia_css_frame_descriptor;
-
-	struct ipu_fw_psys_stream *ipu_fw_psys_stream;
-	struct ia_css_stream_s *ia_css_stream;
-
-	SIZE_OF_CHECK(ipu_fw_psys_pg, ia_css_pg);
-	ABI_CHECK(ipu_fw_psys_pg, ia_css_pg, ID);
-	ABI_CHECK(ipu_fw_psys_pg, ia_css_pg, process_count);
-	ABI_CHECK(ipu_fw_psys_pg, ia_css_pg, processes_offset);
-	ABI_CHECK(ipu_fw_psys_pg, ia_css_pg, routing_bitmap);
-
-	SIZE_OF_CHECK(ipu_fw_psys_ps, ia_css_ps);
-	ABI_CHECK(ipu_fw_psys_ps, ia_css_ps, ID);
-	ABI_CHECK(ipu_fw_psys_ps, ia_css_ps, dev_chn_offset);
-	ABI_CHECK(ipu_fw_psys_ps, ia_css_ps, cell_id);
-
-	SIZE_OF_CHECK(ipu_fw_psys_init, ia_css_psys_init);
-	ABI_CHECK(ipu_fw_psys_init, ia_css_psys_init, icache_prefetch_sp);
-	ABI_CHECK(ipu_fw_psys_init, ia_css_psys_init, icache_prefetch_isp);
-
-	SIZE_OF_CHECK(ipu_fw_psys_cmd, ia_css_psys_cmd);
-	ABI_CHECK(ipu_fw_psys_cmd, ia_css_psys_cmd, command);
-	ABI_CHECK(ipu_fw_psys_cmd, ia_css_psys_cmd, msg);
-	ABI_CHECK(ipu_fw_psys_cmd, ia_css_psys_cmd, context_handle);
-
-	SIZE_OF_CHECK(ipu_fw_psys_event, ia_css_psys_event);
-	ABI_CHECK(ipu_fw_psys_event, ia_css_psys_event, status);
-	ABI_CHECK(ipu_fw_psys_event, ia_css_psys_event, token);
-
-	SIZE_OF_CHECK(ipu_fw_psys_terminal, ia_css_terminal);
-	ABI_CHECK(ipu_fw_psys_terminal, ia_css_terminal, terminal_type);
-
-	SIZE_OF_CHECK(ipu_fw_psys_param_terminal, ia_css_param_terminal);
-	ABI_CHECK(ipu_fw_psys_param_terminal, ia_css_param_terminal,
-		  param_payload);
-
-	SIZE_OF_CHECK(ipu_fw_psys_param_payload, ia_css_param_payload);
-	ABI_CHECK(ipu_fw_psys_param_payload, ia_css_param_payload, buffer);
-
-	SIZE_OF_CHECK(ipu_fw_psys_data_terminal, ia_css_data_terminal);
-	ABI_CHECK(ipu_fw_psys_data_terminal, ia_css_data_terminal, frame);
-	ABI_CHECK(ipu_fw_psys_data_terminal, ia_css_data_terminal,
-		  connection_type);
-
-	SIZE_OF_CHECK(ipu_fw_psys_frame, ia_css_frame);
-	ABI_CHECK(ipu_fw_psys_frame, ia_css_frame, data_bytes);
-	ABI_CHECK(ipu_fw_psys_frame, ia_css_frame, data);
-	ABI_CHECK(ipu_fw_psys_frame, ia_css_frame, buffer_state);
-
-	SIZE_OF_CHECK(ipu_fw_psys_frame_descriptor, ia_css_frame_descriptor);
-	SIZE_OF_CHECK(ipu_fw_psys_stream, ia_css_stream);
-}
+#include <ia_css_psys_process_private_types.h>
 
 int ipu_fw_psys_pg_start(struct ipu_psys_kcmd *kcmd)
 {
@@ -534,15 +420,15 @@ static int __init libcsspsys2600_init(void)
 	if (!syscom_buffer)
 		return -ENOMEM;
 
-	syscom_config = kzalloc(
-		sizeof(struct ia_css_syscom_config), GFP_KERNEL);
+	syscom_config = kzalloc(sizeof(struct ia_css_syscom_config),
+				GFP_KERNEL);
 	if (!syscom_config) {
 		rval = -ENOMEM;
 		goto out_syscom_buffer_free;
 	}
 
-	server_init = kzalloc(
-		sizeof(struct ia_css_psys_server_init), GFP_KERNEL);
+	server_init = kzalloc(sizeof(struct ia_css_psys_server_init),
+			      GFP_KERNEL);
 	if (!server_init) {
 		rval = -ENOMEM;
 		goto out_syscom_config_free;
@@ -554,17 +440,13 @@ static int __init libcsspsys2600_init(void)
 
 	*syscom_config = *ia_css_psys_specify();
 	syscom_config->specific_addr = server_init;
-	syscom_config->specific_size =
-		sizeof(struct ia_css_psys_server_init);
+	syscom_config->specific_size = sizeof(struct ia_css_psys_server_init);
 	syscom_config->ssid = PSYS_SSID;
 	syscom_config->mmid = PSYS_MMID;
-	syscom_config->regs_addr =
-		ipu_device_cell_memory_address(
-			SPC0, IPU_DEVICE_SP2600_CONTROL_REGS);
-	syscom_config->dmem_addr =
-		ipu_device_cell_memory_address(
-			SPC0, IPU_DEVICE_SP2600_CONTROL_DMEM);
-	abi_sanity_checker();
+	syscom_config->regs_addr = ipu_device_cell_memory_address(SPC0,
+					IPU_DEVICE_SP2600_CONTROL_REGS);
+	syscom_config->dmem_addr = ipu_device_cell_memory_address(SPC0,
+					IPU_DEVICE_SP2600_CONTROL_DMEM);
 
 	return 0;
 
diff --git a/drivers/media/pci/intel/ipu4/ipu4-css/libintel-ipu4.c b/drivers/media/pci/intel/ipu4/ipu4-css/libintel-ipu4.c
index 74047ca..98ad22a 100644
--- a/drivers/media/pci/intel/ipu4/ipu4-css/libintel-ipu4.c
+++ b/drivers/media/pci/intel/ipu4/ipu4-css/libintel-ipu4.c
@@ -7,7 +7,6 @@
 #include "ipu-isys.h"
 #include "ipu-wrapper.h"
 #include <ia_css_isysapi.h>
-#include "libintel-checker.h"
 
 #define ipu_lib_call_notrace_unlocked(func, isys, ...)		\
 	({								\
@@ -387,18 +386,5 @@ int ipu_fw_isys_complex_cmd(struct ipu_isys *isys,
 }
 EXPORT_SYMBOL_GPL(ipu_fw_isys_complex_cmd);
 
-static int __init library_init(void)
-{
-	ipu_isys_abi_checker();
-	return 0;
-}
-
-static void __exit library_exit(void)
-{
-}
-
-module_init(library_init);
-module_exit(library_exit);
-
 MODULE_LICENSE("GPL");
 MODULE_DESCRIPTION("Intel ipu library");
diff --git a/drivers/media/pci/intel/ipu4/ipu4p-css/Makefile.isyslib b/drivers/media/pci/intel/ipu4/ipu4p-css/Makefile.isyslib
index 0f066d9..f0b540d 100644
--- a/drivers/media/pci/intel/ipu4/ipu4p-css/Makefile.isyslib
+++ b/drivers/media/pci/intel/ipu4/ipu4p-css/Makefile.isyslib
@@ -17,7 +17,6 @@ $(shell cp -f $(srcpath)/$(src)/../ipu-wrapper.c $(srcpath)/$(src)/ipu4p-css/ipu
 
 intel-ipu4p-isys-csslib-objs := \
 		ipu4p-css/libintel-ipu4p.o \
-		../libintel-checker.o \
 		$(IPU_ISYSLIB_SRC)
 
 ifeq ($(CONFIG_VIDEO_INTEL_IPU), m)
diff --git a/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600psys/libcsspsys2600.c b/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600psys/libcsspsys2600.c
index d9b4661..af0e41e 100644
--- a/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600psys/libcsspsys2600.c
+++ b/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600psys/libcsspsys2600.c
@@ -13,135 +13,21 @@
 
 #include <linux/delay.h>
 #include <linux/module.h>
-#include "libcsspsys2600.h"
-#include <vied_nci_psys_resource_model.h>
-#include <ia_css_psys_device.h>
-#include <ipu_device_cell_properties_func.h>
 
 #include <uapi/linux/ipu-psys.h>
 
 #include "ipu.h"
-#include "ipu-fw-psys.h"
+#include "ipu-mmu.h"
 #include "ipu-psys.h"
 #include "ipu-wrapper.h"
-#include "ipu-mmu.h"
+#include "ipu-fw-psys.h"
+#include "libcsspsys2600.h"
 
-#include <ia_css_psys_process_group_cmd_impl.h>
-#include <ia_css_psys_process_private_types.h>
-#include <ia_css_psys_init.h>
-#include <ia_css_psys_transport.h>
-#include <ia_css_terminal_base_types.h>
-#include <ia_css_terminal_types.h>
-#include <ia_css_program_group_param_types.h>
-#include <ia_css_psys_terminal_private_types.h>
-#include <ia_css_program_group_data.h>
+#include <vied_nci_psys_resource_model.h>
+#include <ia_css_psys_device.h>
+#include <ipu_device_cell_properties_func.h>
 #include <ia_css_psys_program_group_private.h>
-
-#define BASIC_ABI_CHECK
-
-#ifndef BASIC_ABI_CHECK
-#define ABI_CHECK(a, b, field)					\
-	{								\
-	if (offsetof(typeof(*a), field) != offsetof(typeof(*b), field))	\
-		pr_err("intel_ipu4 psys ABI mismatch %s\n",		\
-		       __stringify(field));		    \
-	}
-#else
-#define ABI_CHECK(a, b, field) { }
-#endif
-
-#define SIZE_OF_CHECK(a, b) \
-	{		    \
-	if (sizeof(*a) != sizeof(*b))\
-		pr_err("intel_ipu4 psys ABI size of mismatch %s\n",	\
-		       __stringify(a));					\
-	}								\
-
-static void abi_sanity_checker(void)
-{
-	struct ipu_fw_psys_process_group *ipu_fw_psys_pg;
-	struct ia_css_process_group_s *ia_css_pg;
-
-	struct ipu_fw_psys_process *ipu_fw_psys_ps;
-	struct ia_css_process_s *ia_css_ps;
-
-	struct ipu_fw_psys_srv_init *ipu_fw_psys_init;
-	struct ia_css_psys_server_init *ia_css_psys_init;
-
-	struct ipu_fw_psys_cmd *ipu_fw_psys_cmd;
-	struct ia_css_psys_cmd_s *ia_css_psys_cmd;
-
-	struct ipu_fw_psys_event *ipu_fw_psys_event;
-	struct ia_css_psys_event_s *ia_css_psys_event;
-
-	struct ipu_fw_psys_terminal *ipu_fw_psys_terminal;
-	struct ia_css_terminal_s *ia_css_terminal;
-
-	struct ipu_fw_psys_param_terminal *ipu_fw_psys_param_terminal;
-	struct ia_css_param_terminal_s *ia_css_param_terminal;
-
-	struct ipu_fw_psys_param_payload *ipu_fw_psys_param_payload;
-	struct ia_css_param_payload_s *ia_css_param_payload;
-
-	struct ipu_fw_psys_data_terminal *ipu_fw_psys_data_terminal;
-	struct ia_css_data_terminal_s *ia_css_data_terminal;
-
-	struct ipu_fw_psys_frame *ipu_fw_psys_frame;
-	struct ia_css_frame_s *ia_css_frame;
-
-	struct ipu_fw_psys_frame_descriptor *ipu_fw_psys_frame_descriptor;
-	struct ia_css_frame_descriptor_s *ia_css_frame_descriptor;
-
-	struct ipu_fw_psys_stream *ipu_fw_psys_stream;
-	struct ia_css_stream_s *ia_css_stream;
-
-	SIZE_OF_CHECK(ipu_fw_psys_pg, ia_css_pg);
-	ABI_CHECK(ipu_fw_psys_pg, ia_css_pg, ID);
-	ABI_CHECK(ipu_fw_psys_pg, ia_css_pg, process_count);
-	ABI_CHECK(ipu_fw_psys_pg, ia_css_pg, processes_offset);
-	ABI_CHECK(ipu_fw_psys_pg, ia_css_pg, routing_bitmap);
-
-	SIZE_OF_CHECK(ipu_fw_psys_ps, ia_css_ps);
-	ABI_CHECK(ipu_fw_psys_ps, ia_css_ps, ID);
-	ABI_CHECK(ipu_fw_psys_ps, ia_css_ps, dev_chn_offset);
-	ABI_CHECK(ipu_fw_psys_ps, ia_css_ps, cell_id);
-
-	SIZE_OF_CHECK(ipu_fw_psys_init, ia_css_psys_init);
-	ABI_CHECK(ipu_fw_psys_init, ia_css_psys_init, icache_prefetch_sp);
-	ABI_CHECK(ipu_fw_psys_init, ia_css_psys_init, icache_prefetch_isp);
-
-	SIZE_OF_CHECK(ipu_fw_psys_cmd, ia_css_psys_cmd);
-	ABI_CHECK(ipu_fw_psys_cmd, ia_css_psys_cmd, command);
-	ABI_CHECK(ipu_fw_psys_cmd, ia_css_psys_cmd, msg);
-	ABI_CHECK(ipu_fw_psys_cmd, ia_css_psys_cmd, context_handle);
-
-	SIZE_OF_CHECK(ipu_fw_psys_event, ia_css_psys_event);
-	ABI_CHECK(ipu_fw_psys_event, ia_css_psys_event, status);
-	ABI_CHECK(ipu_fw_psys_event, ia_css_psys_event, token);
-
-	SIZE_OF_CHECK(ipu_fw_psys_terminal, ia_css_terminal);
-	ABI_CHECK(ipu_fw_psys_terminal, ia_css_terminal, terminal_type);
-
-	SIZE_OF_CHECK(ipu_fw_psys_param_terminal, ia_css_param_terminal);
-	ABI_CHECK(ipu_fw_psys_param_terminal, ia_css_param_terminal,
-		  param_payload);
-
-	SIZE_OF_CHECK(ipu_fw_psys_param_payload, ia_css_param_payload);
-	ABI_CHECK(ipu_fw_psys_param_payload, ia_css_param_payload, buffer);
-
-	SIZE_OF_CHECK(ipu_fw_psys_data_terminal, ia_css_data_terminal);
-	ABI_CHECK(ipu_fw_psys_data_terminal, ia_css_data_terminal, frame);
-	ABI_CHECK(ipu_fw_psys_data_terminal, ia_css_data_terminal,
-		  connection_type);
-
-	SIZE_OF_CHECK(ipu_fw_psys_frame, ia_css_frame);
-	ABI_CHECK(ipu_fw_psys_frame, ia_css_frame, data_bytes);
-	ABI_CHECK(ipu_fw_psys_frame, ia_css_frame, data);
-	ABI_CHECK(ipu_fw_psys_frame, ia_css_frame, buffer_state);
-
-	SIZE_OF_CHECK(ipu_fw_psys_frame_descriptor, ia_css_frame_descriptor);
-	SIZE_OF_CHECK(ipu_fw_psys_stream, ia_css_stream);
-}
+#include <ia_css_psys_process_private_types.h>
 
 int ipu_fw_psys_pg_start(struct ipu_psys_kcmd *kcmd)
 {
@@ -542,15 +428,15 @@ static int __init libcsspsys2600_init(void)
 	if (!syscom_buffer)
 		return -ENOMEM;
 
-	syscom_config = kzalloc(
-		sizeof(struct ia_css_syscom_config), GFP_KERNEL);
+	syscom_config = kzalloc(sizeof(struct ia_css_syscom_config),
+				GFP_KERNEL);
 	if (!syscom_config) {
 		rval = -ENOMEM;
 		goto out_syscom_buffer_free;
 	}
 
-	server_init = kzalloc(
-		sizeof(struct ia_css_psys_server_init), GFP_KERNEL);
+	server_init = kzalloc(sizeof(struct ia_css_psys_server_init),
+			      GFP_KERNEL);
 	if (!server_init) {
 		rval = -ENOMEM;
 		goto out_syscom_config_free;
@@ -562,17 +448,13 @@ static int __init libcsspsys2600_init(void)
 
 	*syscom_config = *ia_css_psys_specify();
 	syscom_config->specific_addr = server_init;
-	syscom_config->specific_size =
-		sizeof(struct ia_css_psys_server_init);
+	syscom_config->specific_size = sizeof(struct ia_css_psys_server_init);
 	syscom_config->ssid = PSYS_SSID;
 	syscom_config->mmid = PSYS_MMID;
-	syscom_config->regs_addr =
-		ipu_device_cell_memory_address(
-			SPC0, IPU_DEVICE_SP2600_CONTROL_REGS);
-	syscom_config->dmem_addr =
-		ipu_device_cell_memory_address(
-			SPC0, IPU_DEVICE_SP2600_CONTROL_DMEM);
-	abi_sanity_checker();
+	syscom_config->regs_addr = ipu_device_cell_memory_address(SPC0,
+					IPU_DEVICE_SP2600_CONTROL_REGS);
+	syscom_config->dmem_addr = ipu_device_cell_memory_address(SPC0,
+					IPU_DEVICE_SP2600_CONTROL_DMEM);
 
 	return 0;
 
diff --git a/drivers/media/pci/intel/ipu4/ipu4p-css/libintel-ipu4p.c b/drivers/media/pci/intel/ipu4/ipu4p-css/libintel-ipu4p.c
index 8d62b95..37f84d6 100644
--- a/drivers/media/pci/intel/ipu4/ipu4p-css/libintel-ipu4p.c
+++ b/drivers/media/pci/intel/ipu4/ipu4p-css/libintel-ipu4p.c
@@ -7,7 +7,6 @@
 #include "ipu-isys.h"
 #include "ipu-wrapper.h"
 #include <ia_css_isysapi.h>
-#include "libintel-checker.h"
 
 #define ipu_lib_call_notrace_unlocked(func, isys, ...)		\
 	({								\
@@ -388,18 +387,5 @@ int ipu_fw_isys_complex_cmd(struct ipu_isys *isys,
 }
 EXPORT_SYMBOL_GPL(ipu_fw_isys_complex_cmd);
 
-static int __init library_init(void)
-{
-	ipu_isys_abi_checker();
-	return 0;
-}
-
-static void __exit library_exit(void)
-{
-}
-
-module_init(library_init);
-module_exit(library_exit);
-
 MODULE_LICENSE("GPL");
 MODULE_DESCRIPTION("Intel ipu library");
diff --git a/drivers/media/pci/intel/libintel-checker.c b/drivers/media/pci/intel/libintel-checker.c
deleted file mode 100644
index 7def833..0000000
--- a/drivers/media/pci/intel/libintel-checker.c
+++ /dev/null
@@ -1,165 +0,0 @@
-// SPDX-License_Identifier: GPL-2.0
-// Copyright (C) 2016 - 2018 Intel Corporation
-
-#include <linux/module.h>
-#include <linux/stringify.h>
-#include "libintel-checker.h"
-#include "ipu-fw-isys.h"
-#include "ia_css_isys_fw_bridged_types.h"
-
-/*
- * Comment this out for field-by-field check. When defined only sizeof level
- * check will be done
- */
-#define BASIC_ABI_CHECK
-
-#define ABI_CHECK(a, b, field) { }
-
-#define SIZE_OF_CHECK(a, b) \
-	{		    \
-	if (sizeof(*a) != sizeof(*b))\
-		pr_err("warning: ipu isys ABI size of mismatch %s\n",	\
-		       __stringify(a));					\
-	}								\
-
-static void abi_sanity_checker(void)
-{
-	struct ipu_fw_isys_buffer_partition_abi *abi_buf_part;
-	struct ia_css_isys_buffer_partition_comm *comm_buf_part;
-
-	struct ipu_fw_isys_fw_config *abi_fw_config;
-	struct ia_css_isys_fw_config *comm_fw_config;
-
-	struct ipu_fw_isys_resolution_abi *abi_resol;
-	struct ia_css_isys_resolution_comm *comm_resol;
-
-	struct ipu_fw_isys_output_pin_payload_abi *abi_pin_payload;
-	struct ia_css_isys_output_pin_payload_comm *comm_pin_payload;
-
-	struct ipu_fw_isys_output_pin_info_abi *abi_output_pin_info;
-	struct ia_css_isys_output_pin_info_comm *comm_output_pin_info;
-
-	struct ipu_fw_isys_param_pin_abi *abi_param_pin;
-	struct ia_css_isys_param_pin_comm *comm_param_pin;
-
-	struct ipu_fw_isys_isa_cfg_abi *abi_isa_cfg;
-	struct ia_css_isys_isa_cfg_comm *comm_isa_cfg;
-
-	struct ipu_fw_isys_input_pin_info_abi *abi_input_pin_info;
-	struct ia_css_isys_input_pin_info_comm *comm_input_pin_info;
-
-	struct ipu_fw_isys_cropping_abi *abi_cropping;
-	struct ia_css_isys_cropping_comm *comm_cropping;
-
-	struct ipu_fw_isys_stream_cfg_data_abi *abi_stream_cfg;
-	struct ia_css_isys_stream_cfg_data_comm *comm_stream_cfg;
-
-	struct ipu_fw_isys_frame_buff_set_abi *abi_frame_buff_set;
-	struct ia_css_isys_frame_buff_set_comm *comm_frame_buff_set;
-
-	struct ipu_fw_isys_error_info_abi *abi_err_info;
-	struct ia_css_isys_error_info_comm *comm_err_info;
-
-	struct ipu_fw_isys_resp_info_abi *abi_resp;
-	struct ia_css_isys_resp_info_comm *comm_resp;
-
-	struct ipu_fw_isys_proxy_error_info_abi *abi_proxy_err;
-	struct ia_css_isys_proxy_error_info_comm *comm_proxy_err;
-
-	struct ipu_fw_isys_proxy_resp_info_abi *abi_proxy_resp;
-	struct ia_css_isys_proxy_resp_info_comm *comm_proxy_resp;
-
-	struct ipu_fw_proxy_write_queue_token *abi_write_token;
-	struct ia_css_proxy_write_queue_token *comm_write_token;
-
-	struct ipu_fw_resp_queue_token *abi_resp_token;
-	struct resp_queue_token *comm_resp_token;
-
-	struct ipu_fw_send_queue_token *abi_send_token;
-	struct send_queue_token *comm_send_token;
-
-	struct ipu_fw_proxy_resp_queue_token *abi_resp_queue;
-	struct proxy_resp_queue_token *comm_resp_queue;
-
-	struct ipu_fw_proxy_send_queue_token *abi_proxy_send;
-	struct proxy_send_queue_token *comm_proxy_send;
-
-	SIZE_OF_CHECK(abi_buf_part, comm_buf_part);
-	SIZE_OF_CHECK(abi_fw_config, comm_fw_config);
-	SIZE_OF_CHECK(abi_err_info, comm_err_info);
-	SIZE_OF_CHECK(abi_proxy_err, comm_proxy_err);
-	SIZE_OF_CHECK(abi_proxy_resp, comm_proxy_resp);
-	SIZE_OF_CHECK(abi_write_token, comm_write_token);
-	SIZE_OF_CHECK(abi_resp_token, comm_resp_token);
-	SIZE_OF_CHECK(abi_send_token, comm_send_token);
-	SIZE_OF_CHECK(abi_resp_queue, comm_resp_queue);
-	SIZE_OF_CHECK(abi_proxy_send, comm_proxy_send);
-
-	SIZE_OF_CHECK(abi_resp, comm_resp);
-	ABI_CHECK(abi_resp, comm_resp, buf_id);
-	ABI_CHECK(abi_resp, comm_resp, type);
-	ABI_CHECK(abi_resp, comm_resp, timestamp[0]);
-	ABI_CHECK(abi_resp, comm_resp, timestamp[1]);
-	ABI_CHECK(abi_resp, comm_resp, stream_handle);
-	ABI_CHECK(abi_resp, comm_resp, error_info.error);
-	ABI_CHECK(abi_resp, comm_resp, error_info.error_details);
-	ABI_CHECK(abi_resp, comm_resp, pin.out_buf_id);
-	ABI_CHECK(abi_resp, comm_resp, pin.addr);
-	ABI_CHECK(abi_resp, comm_resp, pin_id);
-	ABI_CHECK(abi_resp, comm_resp, process_group_light.param_buf_id);
-	ABI_CHECK(abi_resp, comm_resp, process_group_light.addr);
-	ABI_CHECK(abi_resp, comm_resp, acc_id);
-
-	SIZE_OF_CHECK(abi_resol, comm_resol);
-	ABI_CHECK(abi_resol, comm_resol, width);
-	ABI_CHECK(abi_resol, comm_resol, height);
-
-	SIZE_OF_CHECK(abi_pin_payload, comm_pin_payload);
-	ABI_CHECK(abi_pin_payload, comm_pin_payload, out_buf_id);
-	ABI_CHECK(abi_pin_payload, comm_pin_payload, addr);
-
-	SIZE_OF_CHECK(abi_output_pin_info, comm_output_pin_info);
-	ABI_CHECK(abi_output_pin_info, comm_output_pin_info, input_pin_id);
-	ABI_CHECK(abi_output_pin_info, comm_output_pin_info, stride);
-	ABI_CHECK(abi_output_pin_info, comm_output_pin_info, pt);
-	ABI_CHECK(abi_output_pin_info, comm_output_pin_info,
-		  watermark_in_lines);
-	ABI_CHECK(abi_output_pin_info, comm_output_pin_info, send_irq);
-	ABI_CHECK(abi_output_pin_info, comm_output_pin_info, ft);
-
-	SIZE_OF_CHECK(abi_param_pin, comm_param_pin);
-	ABI_CHECK(abi_param_pin, comm_param_pin, param_buf_id);
-	ABI_CHECK(abi_param_pin, comm_param_pin, addr);
-
-	SIZE_OF_CHECK(abi_isa_cfg, comm_isa_cfg);
-
-	SIZE_OF_CHECK(abi_input_pin_info, comm_input_pin_info);
-	ABI_CHECK(abi_input_pin_info, comm_input_pin_info, dt);
-	ABI_CHECK(abi_input_pin_info, comm_input_pin_info, mipi_store_mode);
-
-	SIZE_OF_CHECK(abi_cropping, comm_cropping);
-	ABI_CHECK(abi_cropping, comm_cropping, top_offset);
-	ABI_CHECK(abi_cropping, comm_cropping, left_offset);
-	ABI_CHECK(abi_cropping, comm_cropping, bottom_offset);
-	ABI_CHECK(abi_cropping, comm_cropping, right_offset);
-
-	SIZE_OF_CHECK(abi_stream_cfg, comm_stream_cfg);
-	ABI_CHECK(abi_stream_cfg, comm_stream_cfg, src);
-	ABI_CHECK(abi_stream_cfg, comm_stream_cfg, vc);
-	ABI_CHECK(abi_stream_cfg, comm_stream_cfg, isl_use);
-	ABI_CHECK(abi_stream_cfg, comm_stream_cfg, compfmt);
-	ABI_CHECK(abi_stream_cfg, comm_stream_cfg, send_irq_sof_discarded);
-	ABI_CHECK(abi_stream_cfg, comm_stream_cfg, send_irq_eof_discarded);
-	ABI_CHECK(abi_stream_cfg, comm_stream_cfg, nof_input_pins);
-	ABI_CHECK(abi_stream_cfg, comm_stream_cfg, nof_output_pins);
-	ABI_CHECK(abi_stream_cfg, comm_stream_cfg, nof_input_pins);
-
-	SIZE_OF_CHECK(abi_frame_buff_set, comm_frame_buff_set);
-	ABI_CHECK(abi_frame_buff_set, abi_frame_buff_set, send_irq_sof);
-	ABI_CHECK(abi_frame_buff_set, abi_frame_buff_set, send_irq_eof);
-}
-
-void ipu_isys_abi_checker(void)
-{
-	abi_sanity_checker();
-}
diff --git a/drivers/media/pci/intel/libintel-checker.h b/drivers/media/pci/intel/libintel-checker.h
deleted file mode 100644
index 83a068a..0000000
--- a/drivers/media/pci/intel/libintel-checker.h
+++ /dev/null
@@ -1,9 +0,0 @@
-/* SPDX-License_Identifier: GPL-2.0 */
-/* Copyright (C) 2016 - 2018 Intel Corporation */
-
-#ifndef LIBINTEL_CHECKER_H
-#define LIBINTEL_CHECKER_H
-
-void ipu_isys_abi_checker(void);
-
-#endif
-- 
1.9.1

